### PR TITLE
Fix comparison warning in lod_reset_op.h

### PR DESCRIPTION
**Pull request by abhinavarora**
Fixes https://github.com/PaddlePaddle/Paddle/issues/9672

### CODE REVIEW

File: paddle/fluid/operators/lod_reset_op.h [hunk 1]
```diff
@@ -35,7 +37,7 @@ class LoDResetKernel : public framework::OpKernel<T> {
       if (lod_t->lod().size() > 0) {
         auto y_lod = lod_t->lod();
         auto last_level = y_lod[y_lod.size() - 1];
-        PADDLE_ENFORCE_EQ(last_level.back(), in->dims()[0],
+        PADDLE_ENFORCE_EQ((int64_t)(last_level.back()), in->dims()[0],
```
Comment (wangkuiyi) [inline paddle/fluid/operators/lod_reset_op.h]: Would cpplint complain about this C-style cast?

Comment (abhinavarora) [reply paddle/fluid/operators/lod_reset_op.h]: No, it doesn't

Comment (abhinavarora) [reply paddle/fluid/operators/lod_reset_op.h]: Do you think I should still change it to static_cast

Comment (wangkuiyi) [reply paddle/fluid/operators/lod_reset_op.h]: It is curious that cpplint doesn't complain. But as it doesn't, I think this PR is great! Thanks!

**Pull request merged**